src/pr-allowed.js

```diff
@@ -0,0 +1,78 @@
+let yaml = require('js-yaml');
+
+const DEFAULT_POLICY = 'collaborators';
+
+async function prAllowed(options) {
+  switch (await getRepoPolicy(options)) {
+    case 'collaborators':
+      return await isCollaborator(options);
+
+    case 'public':
+      return true;
+
+    default:
+      return false;
+  }
+}
+
+/**
+ * Get the repo's "policy" on pull requests, by fetching .taskcluster.yml from the default
+ * branch, parsing it, and looking at its `allowPullRequests`.
+ */
+async function getRepoPolicy({login, organization, repository, instGithub, debug}) {
+  // first, get the repository's default branch
+  let repoInfo = await instGithub.repos.get({owner: organization, repo: repository});
+  let branch = repoInfo.default_branch;
+
+  // load .taskcluster.yml from that branch
+  let taskclusterYml;
+  try {
+    let content = await instGithub.repos.getContent({
+      owner: organization,
+      repo: repository,
+      path: '.taskcluster.yml',
+      ref: branch,
+    });
+    taskclusterYml = yaml.safeLoad(new Buffer(content.content, 'base64').toString());
+  } catch (e) {
+    if (e.code === 404) {
+      return DEFAULT_POLICY;
+    }
+    throw e;
+  }
+
+  // consult its `allowPullRequests` field
+  return taskclusterYml['allowPullRequests'] || DEFAULT_POLICY;
+}
+
+async function isCollaborator({login, organization, repository, sha, instGithub, debug}) {
+  if (login === organization) {
```

Comment:
I don't think this is necessary anymore. The owner should be a collaborator according to their docs. (and also now according to me trying it locally)